readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/HeartbeatConsumer.java

```diff
@@ -41,18 +41,7 @@ public HeartbeatConsumer(Collection<PartitionedWriter<MESSAGE_KIND>> writers, Te
     public void start(Thread.UncaughtExceptionHandler uncaughtExceptionHandler, String name) {
         runningThread = new Thread(() -> {
             while (!Thread.currentThread().isInterrupted()) {
-                synchronized (latestPartitionsOffset) {
-                    for (Map.Entry<Integer, Offset> partitionOffset : latestPartitionsOffset.entrySet()) {
-                        int partition = partitionOffset.getKey();
-                        Offset offset = partitionOffset.getValue();
-                        Offset latestHeartbeat = latestHeartbeats.get(partition);
-
-                        if (!offset.equals(latestHeartbeat)) {
-                            latestHeartbeats.put(partition, offset);
-                            writers.forEach(writer -> writer.heartbeat(partition, offset));
-                        }
-                    }
-                }
+                run();
```

Comment:
I thought this commit would avoid running HeartbeatConsumer and Expirer in dedicated threads. Did I miss something ?

Reply:
done